### ISSUE TITLE
Tabs Design Manual page

### ISSUE THREAD

**Issue by Dnpizarro** [cfpb/design-manual#352]
Working from @keelerr [outline](https://docs.google.com/document/d/1_nkqDqFYu2RnwbiR1w2TcRRNIjeC8t-68-s18Z2w4mg/edit) for the Tabs design manual page, I'll start mocking up the visuals and laying out the [content](https://docs.google.com/document/d/1m1UIb_DplE7i719rwozFb6vVGmOmyLwYFkmYMNyazAc/edit#).

Stay posted on page designs....

**Comment by Dnpizarro** [comment]
@keelerr here is the most recent mock-up I showed during our Working Session. I'll keep working on the design, while you try to shorten some of the bullet points. Also, for the link under the image, should we show the full url or just the label (i.e. Form explainer)? I believe our standard is to show the url so that the user knows where they are going. Also on that note, if we decide to show the url I'm not sure if we need to add the external link minicon...

![ui-toolkit_tabs_dm_page_r1_tabs-01_tabs-01](https://cloud.githubusercontent.com/assets/6852363/9590223/4bad7d24-5001-11e5-96de-59ab39179bf4.png)

**Comment by Dnpizarro** [comment]
@keelerr the designs below include all of the content provided in [outline document](https://docs.google.com/document/d/1_nkqDqFYu2RnwbiR1w2TcRRNIjeC8t-68-s18Z2w4mg/edit#heading=h.cx1kf2t0uhgz). I made some suggestions to simplify a couple of the headers ( "Tab usage" = "Usage", "1. To show alternate..." = "Use case 1: Show alternate..."). There are still some areas that could use some cutting back on content, but I wanted to show this full view as you edit it down. Also, one thing that I just realized is that I should include some illustrated measurements for how the tabs should be styled (i.e. height of tab, padding, etc)

![ui-toolkit_tabs_dm_page_tabs-01](https://cloud.githubusercontent.com/assets/6852363/9614182/26a8651c-50be-11e5-9029-fa628dba5802.png)

**Comment by kurzn** [comment]
I just ran across this implementation of tabs on Paying for College: expandables as tabbed interaction. Can we add a comment in the tabs rules about NOT doing this, please?   

http://www.consumerfinance.gov/paying-for-college/choose-a-student-loan/#o1 

![image](https://cloud.githubusercontent.com/assets/10562534/9618367/97d62474-50cb-11e5-84b3-cf4a7e74564b.png)

**Comment by benguhin** [comment]
@kurzn I totally agree. That said, we could have very long pages if we try to detail everything that people should _not_ do :)

I think the rules for the [expandable sections](https://github.com/cfpb/design-manual/issues/143) pattern will help make sure that that interaction doesn't happen again

**Comment by benguhin** [comment]
Similar to @Scotchester's [comment about tooltips](https://github.com/cfpb/design-manual/issues/139#issuecomment-136483183), I don't think we should standardize the use of minicons instead of text for the name of each tab. Iconography and hover states are not clear enough indicators of what's underneath the tab.

**Comment by Dnpizarro** [comment]
### Latest update for Friday Presentation

![ui-toolkit_tabs_dm_page_r2_tabs](https://cloud.githubusercontent.com/assets/6852363/9690759/5aa0e72c-530c-11e5-8eae-826f24ce5955.png)

**Comment by nataliafitzgerald** [comment]
@Dnpizarro @keelerr  - I don't think that I agree with including the minicon tabs as a standard. Also, the OaH tabs are taller than the standard you have set forth. So, for the image I would rather see tabs that are following the guidelines you are proposing. If you'd like to chat through this I'm available.

**Comment by Dnpizarro** [comment]
### Update 9/14

@nataliafitzgerald sorry for not updating this issue sooner, but we are definitely not including the minicon tab as a standard. In the mockup below, under the section "Use" I plan on showing live versions of the tabs. I included some lorem ipsum to fill in the example. Also, I included some questions in the mockup, particularly the one about whether we should consider numbering the tabs in the "Navigate a process" example. 

![ui-toolkit_tabs_dm_page_r3_tabs](https://cloud.githubusercontent.com/assets/6852363/9855904/3aca2a32-5adf-11e5-8bf4-32fd1b585f7a.png)

**Comment by nataliafitzgerald** [comment]
- It looks like we are repeating the content between the introduction and the paragraph under "Use." The text seems to be more applicable to the "Use" section. 
- Also, you mention that we would "rarely" use tabs to navigate a process. The explanation is pretty complicated. Is there really an essential need for this tab usage? 
- Wording edit: If we are thinking of the word "Use" as the first word in the sentence, then the titles for the two use cases would be: "1. To show alternate views of the same information" and "2. To navigate through a process" OR you can remove the word "To" from both. 
- It would be useful to include examples of use cases for each of the tab types. These don't have to be visualized they could just be mentioned in the text. 
- The black minicon numbers feel too heavy (and distracts from the tab header). And I wonder whether including them would preclude a designer from instead doing something like "Step 1" "Step 2"
- It looks like there is a typo in the first bullet under "Technical considerations"
- For the layout, make sure to follow the current standard for the Design Manual pages. The text under "Use" should be Georgia and the intro text should be larger.

**Comment by keelerr** [comment]
@nataliafitzgerald and @Dnpizarro, I've tried to respond to your points:
1. I think the introductory sentence should stay in the introduction. I don't think we need any more preamble before jumping into the use case. (And this is what's included in my current draft of the content.)
2. I think using tabs to step users through a process is a valid use case. This is why I've included it. To help me clarify and improve this section, can you tell me what seems complicated about it?
3. Thanks for the wording suggestion. I like including "to" in each use case. So, it would be "To show alternate views of the same information" and "To navigate through a process."
4. I think it would be more helpful to include examples rather than the generic "tab 1" and "tab 2." Is there a reason we've moved away from the CFPB examples? I think it provides helpful context. And, the  example link you're showing under "To show alternate views..." is really to illustrate "To navigate through a process". This might help clarify @nataliafitzgerald's concern about the 2nd use case. If they didn't look identical, it might be easier to distinguish why there are 2 use cases. 
5. I agree the black is heavy, but I think including "Step 1" and "Step 2" may also become problematic. With tight character counts, this may not leave enough space to include a descriptive word/phrase explaining what the step is!  @Dnpizarro, I think it's helpful to include numbering. I don't think we have to mandate it be included, but I'd love the example to show some numbering...
6. Do we just want to eliminate this section all together? (It should read: Use name anchors. Consider...)

**Comment by nataliafitzgerald** [comment]
@Dnpizarro - If you include numbers you can look at a different coloring for the minicon circle or use numbers without circles.

**Comment by Dnpizarro** [comment]
I started coding the DM page, see below. I will post screenshots of the examples tomorrow. @keelerr not sure were we landed on the "Technical considerations," did we still want to include that? Also, the links connecting to the code still need to be corrected. @Scotchester would you be able to provide those links for me?

http://dnpizarro.github.io/design-manual/ui-toolkit/tabs.html

**Comment by Dnpizarro** [comment]
@keelerr and @nataliafitzgerald what are your thoughts on having the example go full width in the main content area after the use case explanation. The bullet points would go after the example. Additionally, @keelerr and I discussed the possibility of defining the bottom of the tab with a rule line. For clarity, the rule just above the bullets is the tab's bottom rule line. An alternate option would be to just switch the example with the bullets. Let me know your thoughts.

FYI, I cropped it intentionally to highlight the use case examples.

![ui-toolkit_tabs_dm_page_r4_tabs](https://cloud.githubusercontent.com/assets/6852363/10111902/3fc3a076-63a5-11e5-8b6f-f3666fa727f6.png)

**Comment by keelerr** [comment]
@Dnpizarro, in answer to your questions:
- I like the examples going the width of the content area. The only problem is that right now the 2 use cases look identical—other than the use of numbers. Since the second use case is meant to be used for navigation, would a "screenshot" of tabs spanning the width of the page be more helpful? I don't think this example really captures the use case. And, could we pair it with a link to the OAH journey to show people a live example? 
- I think it might be helpful to switch the bullets with the example. This way, all the text would be together. If this creates weird white spaces before the example and looks weird, than I leave it up to you and @nataliafitzgerald.
- As for technical considerations, I think the one that's most helpful/ worth keeping is the one about named anchors.

**Comment by nataliafitzgerald** [comment]
In terms of "Alternate views of the same information" I can think of a lot of examples of tabs used to show multiple views in tables or charts. Would tables or charts be a good use case for this DM issue or are you looking for a more text based example. 

Here are two examples from Fitbit:

| Weight over time | Steps over time |
| --- | --- |
| <img width="559" alt="screen shot 2015-09-28 at 4 39 38 pm" src="https://cloud.githubusercontent.com/assets/6887128/10148165/0a7f1cee-6600-11e5-9d10-d0cceefbdad5.png"> | <img width="972" alt="screen shot 2015-09-28 at 4 40 53 pm" src="https://cloud.githubusercontent.com/assets/6887128/10148216/2ff84eb4-6600-11e5-95b0-d2a9b23726ff.png"> |

**Comment by KimberlyMunoz** [comment]
Something I just discovered this week is that WAI-ARIA outlines the intended interactions for things like tabs. It's worth reading and perhaps linking to under the technical considerations.

https://www.w3.org/TR/wai-aria-practices-1.1/#tabpanel

We should be programming keyboard interactions to have screenreader users navigate according to the spec.

**Comment by caheberer** [comment]
The recommendation from this is to not use tabs...EVER. So should we even include tabs in the Design Manual?

**Comment by caheberer** [comment]
@keelerr will take a look at the language to reaffirm that we shouldn't use tabs except in extreme cases.
